spec/services/doc_auth_router_spec.rb

```diff
@@ -33,6 +33,107 @@
     end
   end
 
+  describe '.client and .doc_auth_vendor' do
+    context 'with randomize vendor configuration on' do
+      let(:doc_auth_vendor) { 'test1' }
+      let(:doc_auth_vendor_randomize) { true }
+      let(:doc_auth_vendor_randomize_alternate_vendor) { 'test2' }
+      let(:iterations) { 750 }
+      let(:percent_variance) { 0.04 } # 3.5% variance in randomness
+
+      before(:each) do
+        allow(IdentityConfig.store).to receive(:doc_auth_vendor_randomize).
+          and_return(doc_auth_vendor_randomize)
+        allow(IdentityConfig.store).to receive(:doc_auth_vendor_randomize_alternate_vendor).
+          and_return(doc_auth_vendor_randomize_alternate_vendor)
+      end
+
+      it 'doc_auth_vendor randomizes near configured level' do
+        doc_auth_vendor_randomize_percent = 57
+        allow(IdentityConfig.store).to receive(:doc_auth_vendor_randomize_percent).
+          and_return(doc_auth_vendor_randomize_percent)
+
+        results = []
+        (1..iterations).each do |_i|
```

Comment:
Maybe more expressive?

```suggestion
        iterations.times do
```

Reply:
I can never remember all the helpers that ruby has. Thanks, I forgot about that one.

Reply:
I don't think this is worthwhile to test all variations? Like I think it would be more productive to have a fixed thing with a known SHA, and a fixed percentage, maybe then run a test when we want 0, 50, 100% and see where it routes?